### PR TITLE
Circulars - move date selector component outside of route.tsx to its own file

### PR DESCRIPTION
**Pull request by tylerbarna**
Similar to what was done with the DateSelectorButton component or the CircularsIndex, this PR separates the date selector menu into a separate file outside of route.tsx. This should make it somewhat easier to parse the route file by moving a bit over 150 lines to this separate file. 

Only the start and end date parameters are passed as arguments to the component; I wasn't sure if searchParams itself should be passed.

### CODE REVIEW

File: app/routes/_gcn.circulars._archive._index/DateSelectorMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  Grid,
+  Icon,
+  Radio,
+} from '@trussworks/react-uswds'
+import { useState } from 'react'
+
+import DateSelectorButton from './DateSelectorButton'
+import DetailsDropdownContent from '~/components/DetailsDropdownContent'
+
+export default function ({
```
Comment (lpsinger) [inline app/routes/_gcn.circulars._archive._index/DateSelectorMenu.tsx:23]: We tend not to have default exports in library files.

```suggestion
export function DateSelector ({
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/_gcn.circulars._archive._index/DateSelectorMenu.tsx (hunk 1 above)
Comment (tylerbarna) [reply app/routes/_gcn.circulars._archive._index/DateSelectorMenu.tsx]: Ah, I was taking a cue from the way the DateSelectorButton file was written

**Comment by lpsinger** [comment]
Prettier is complaining about something. What is it upset about?

**Comment by tylerbarna** [comment]
> Prettier is complaining about something. What is it upset about?

I'm trying to figure out what's causing it, Prettier ran fine locally when I made the commit, and the original commit also passed.

**Comment by tylerbarna** [comment]
@lpsinger there was a space in the export function line, should be fixed now

**Review by lpsinger — APPROVED** [review]

**Pull request merged**